packages/eslint-plugin/src/lib/rules/member-naming.ts

```diff
@@ -48,13 +55,16 @@ module.exports = {
     ]
   },
 
-  create(context) {
+  create(context: Rule.RuleContext) {
     const config = util.applyDefault(defaultOptions, context.options)[0];
-    const conventions = Object.keys(config).reduce((acc, accessibility) => {
-      acc[accessibility] = new RegExp(config[accessibility]);
+    const conventions = (Object.keys(config) as Modifiers[]).reduce(
```

Comment:
it would be better to use the strongly typed reduce instead of a cast here, `reduce<Options<RegExp>>`

Reply:
For some reason, I never actually thought about using the generic version of `reduce`. I mean, I knew it existed because otherwise it wouldn't work, I just never thought about using it!